File: apps/ios/GalaxySSI/AgentVoiceAttachmentSubmissionPolicy.swift
import Foundation

enum AgentVoiceAttachmentSubmissionPolicy {
  static func select<T>(
    goalOverride: String?,
    composerAttachments: [T],
    attachmentSnapshot: [T]?
  ) -> [T] {
    attachmentSnapshot ?? (goalOverride == nil ? composerAttachments : [])
  }
}

File: apps/android/app/src/main/java/com/galaxyssi/chat/AgentVoiceAttachmentSubmissionPolicy.kt
package com.galaxyssi.chat

internal object AgentVoiceAttachmentSubmissionPolicy {
    fun <T> select(
        goalOverride: String?,
        composerAttachments: List<T>,
        attachmentSnapshot: List<T>?
    ): List<T> = attachmentSnapshot?.toList()
        ?: if (goalOverride == null) composerAttachments.toList() else emptyList()
}
